Sources/Concurrency/Executor/ConcurrentSequenceExecutor.swift

```diff
@@ -33,8 +33,13 @@ public class ConcurrentSequenceExecutor: SequenceExecutor {
     /// reported error contains the ID of the task that was being executed
     /// when the timeout occurred. The tracking does incur a minor
     /// performance cost. This value defaults to `false`.
-    public init(name: String, qos: DispatchQoS = .userInitiated, shouldTrackTaskId: Bool = false) {
+    /// - parameter maxConcurrentTasks: limits the maximum number of tasks
+    /// run concurrently. Defaults to 0, which will set the max to the number
+    /// of processors.
+    public init(name: String, qos: DispatchQoS = .userInitiated, shouldTrackTaskId: Bool = false, maxConcurrentTasks: Int = 0) {
         taskQueue = DispatchQueue(label: "Executor.taskQueue-\(name)", qos: qos, attributes: .concurrent)
+        let semaphoreCount = maxConcurrentTasks == 0 ? ProcessInfo().processorCount : maxConcurrentTasks
```

Comment:
Can we default this to `-1` there by not restricting at all. GCD already does thread-pooling. From Needle's specific use case, we can pass in processor count x 2, which is probably a good rule of thumb.

Reply:
`-1` would block forever, but we can default it to `INT_MAX` or whatever it is in swift if you prefer

Reply:
By `-1` I just meant some special number that signals no blocking will be performed. But sure, max int works.